dbt-materialize: Support sink cutover in blue/green deployments (#27557)
MaterializeInc/materialize: misc/dbt-materialize/CHANGELOG.md

# dbt-materialize Changelog

## Unreleased

> * Add support for sinks cutover in blue/green deployments [#27557](https://github.com/MaterializeInc/materialize/pull/27557).

Reviewer: ```suggestion
* Add support for sink cutover to the blue/green deployment workflow [#27557](https://github.com/MaterializeInc/materialize/pull/27557).
```

>   Sinks still have to be created in a **dedicated schema and cluster**.

Reviewer: ```suggestion
  Sinks **must** be created in a **dedicated schema and cluster**.
```

> * Introduced `dry_run: True` argument to the `deploy_promote` macro

Reviewer: ```suggestion
* Add a `dry_run` argument to the `deploy_promote` macro, which allows previewing the sequence of commands that will be run as part of the environment promotion step of the blue/green deployment workflow.
```

>   to preview the changes that will be made during the cutover.

Reviewer: ```suggestion
```

> * Added support for `SCHEDULED` cluster type in the `deploy_init` macro.

Reviewer: ```suggestion
* Fix the `deploy_init` macro to correctly account for scheduled clusters. Before, these clusters would be incorrectly recreated in the deployment environment with the `SCHEDULE` option set to `manual` (instead of `on-refresh`.
```
